packages/react-data-grid/src/Canvas.tsx

```diff
@@ -275,47 +313,63 @@ export default function Canvas<R>({
 
   return (
-    <div
-      className="react-grid-Canvas"
-      style={{ height }}
-      ref={canvas}
-      onScroll={handleScroll}
-      onKeyDown={onCanvasKeydown}
-      onKeyUp={onCanvasKeyup}
-    >
-      <InteractionMasks<R>
-        ref={interactionMasks}
-        rowGetter={rowGetter}
-        rowsCount={rowsCount}
-        rowHeight={rowHeight}
-        columns={columnMetrics.columns}
-        height={clientHeight}
-        colVisibleStartIdx={colVisibleStartIdx}
-        colVisibleEndIdx={colVisibleEndIdx}
-        enableCellSelect={enableCellSelect}
-        enableCellAutoFocus={enableCellAutoFocus}
-        cellNavigationMode={cellNavigationMode}
-        eventBus={eventBus}
-        contextMenu={contextMenu}
-        onHitBottomBoundary={onHitBottomCanvas}
-        onHitTopBoundary={onHitTopCanvas}
-        onHitLeftBoundary={handleHitColummBoundary}
-        onHitRightBoundary={handleHitColummBoundary}
-        scrollLeft={scrollLeft}
-        scrollTop={scrollTop}
-        getRowColumns={getRowColumns}
-        editorPortalTarget={editorPortalTarget}
-        {...interactionMasksMetaData}
-      />
-      <RowsContainer id={contextMenu ? contextMenu.props.id : 'rowsContainer'}>
-        <div className="rdg-rows-container" style={{ width: columnMetrics.totalColumnWidth, paddingTop, paddingBottom }}>
-          {getRows()}
+    <>
+      <div
+        className="react-grid-Canvas"
+        style={{ height }}
+        ref={canvas}
+        onScroll={handleScroll}
+        onKeyDown={onCanvasKeydown}
+        onKeyUp={onCanvasKeyup}
+      >
+        <InteractionMasks<R>
+          ref={interactionMasks}
+          rowGetter={rowGetter}
+          rowsCount={rowsCount}
+          rowHeight={rowHeight}
+          columns={columnMetrics.columns}
+          height={clientHeight}
+          colVisibleStartIdx={colVisibleStartIdx}
+          colVisibleEndIdx={colVisibleEndIdx}
+          enableCellSelect={enableCellSelect}
+          enableCellAutoFocus={enableCellAutoFocus}
+          cellNavigationMode={cellNavigationMode}
+          eventBus={eventBus}
+          contextMenu={contextMenu}
+          onHitBottomBoundary={onHitBottomCanvas}
+          onHitTopBoundary={onHitTopCanvas}
+          onHitLeftBoundary={handleHitColummBoundary}
+          onHitRightBoundary={handleHitColummBoundary}
+          scrollLeft={scrollLeft}
+          scrollTop={scrollTop}
+          getRowColumns={getRowColumns}
+          editorPortalTarget={editorPortalTarget}
+          {...interactionMasksMetaData}
+        />
+        <RowsContainer id={contextMenu ? contextMenu.props.id : 'rowsContainer'}>
+          <div className="rdg-rows-container" style={canvasRowsContainerStyle}>
+            {getRows()}
+          </div>
+        </RowsContainer>
+      </div>
+      {summaryRows && summaryRows.length && (
+        <div className="rdg-summary">
+          <div ref={summary} style={boundaryStyle}>
```

Comment:
This is to mimic canvas horizontal scrolling so that for Summary it can just reuse the existing `Row.tsx`.
1. The container is fixed width. so that when this `div` is set the `scrollLeft`, the rows container (with full width, the next line) could be moving around.
2. For sticky frozen divs, we do nothing. For IE 11 specifically, we need to set the `scrollLeft` to each frozen cells. (This logic is just same as what we had)


https://github.com/w3c/csswg-drafts/issues/865
An insteresting finding is that the reason why previously `Header` sticky doesn't work for EDGE is because we set the `.react-grid-HeaderRow` as `overflow: hidden`, it will break the cell position sticky only in EDGE. However, the `overflow: hidden` like the structure of canvas and summary area would work.